### ISSUE TITLE
BUG: skew, kurtosis return np.nan instead of "propagate"

### ISSUE THREAD

**Issue by josef-pkt** [scipy/scipy#5817]
Initial guess

I thought "propagate" just calculates with nans in the arrays, like np.sum, so we get the correct shape with just some extra nans.

Instead we get a scalar nan back
https://github.com/scipy/scipy/blame/v0.17.0/scipy/stats/stats.py#L1319

possible candidate for 
https://github.com/statsmodels/statsmodels/pull/2800

```
>>> aa = np.arange(5 * 4.).reshape(5,4)
>>> aa[0,1] = np.nan
>>> np.sum(aa, 0)
array([ 40.,  nan,  50.,  55.])
>>> from scipy import stats
>>> stats.kurtosis(aa, axis=0)
array([-1.3,  nan, -1.3, -1.3])
>>> stats.skew(aa, axis=0)
array([  0.,  nan,   0.,   0.])
```

```
>>> scipy.__version__
'0.15.1'
```

**Comment by josef-pkt** [comment]
The actual problem for the statsmodels case seems to be scalar versus array return

continued example
1-d versus 2d inputs

```
>>> stats.skew(aa[0], axis=0)
nan
>>> stats.skew(aa[1], axis=0)
0.0

>>> stats.skew(aa[:1], axis=1)
array([ nan])
>>> stats.skew(aa[1:2], axis=1)
array([ 0.])


>>> np.sum(aa[1], axis=0)
22.0
>>> np.sum(aa[1:2], axis=1)
array([ 22.])

>>> np.sum(aa[0], axis=0)
nan
>>> np.sum(aa[:1], axis=1)
array([ nan])
```

**Comment by rgommers** [comment]
That's indeed a regression that needs fixing. I thought we had not merged one PR yet with the functions that suffered from this, but apparently we missed one.

**Comment by josef-pkt** [comment]
This seems to be a pattern that has been added to several `reduce` functions
https://github.com/scipy/scipy/commit/73c4f620bb6602632f6539ca3764a5d8583172b8

I think statsmodels is also using normaltest, skewtest and kurtosistest, but in general we are not supposed to have nans in the calculated data, so there are no broken test cases.

**Comment by ev-br** [comment]
I'll change the milestone to 0.17.1 because this functionality is new in 0.17.0 so it could be good to have it fixed in the 0.17.x series.
